File: backend/app/schemas/analytics.py
from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel


class TypeBreakdownItem(BaseModel):
    interview_type: str
    sessions: int
    avg_score: float


class UserSummaryResponse(BaseModel):
    total_sessions: int
    avg_overall_score: float | None
    avg_technical_score: float | None
    avg_communication_score: float | None
    avg_correctness_score: float | None
    top_weak_topics: list[str]
    by_interview_type: list[TypeBreakdownItem]


class ProgressItem(BaseModel):
    session_id: uuid.UUID
    interview_type: str
    overall_score: float
    completed_at: datetime


class WeakTopicItem(BaseModel):
    topic: str
    count: int

File: frontend/src/types/analytics.ts
import type { InterviewType } from '@/types/interview'

export interface TypeBreakdownItem {
  interview_type: InterviewType
  sessions: number
  avg_score: number
}

/** Returned by GET /analytics/me/summary */
export interface UserSummary {
  total_sessions: number
  avg_overall_score: number | null
  avg_technical_score: number | null
  avg_communication_score: number | null
  avg_correctness_score: number | null
  top_weak_topics: string[]
  by_interview_type: TypeBreakdownItem[]
}

/** Returned by GET /analytics/me/progress (list item) */
export interface ProgressItem {
  session_id: string
  interview_type: InterviewType
  overall_score: number
  completed_at: string // ISO 8601
}

/** Returned by GET /analytics/me/weak-topics (list item) */
export interface WeakTopicItem {
  topic: string
  count: number
}
